### DISCUSSION TITLE
Solidity Version Error while compiling or forge

### DISCUSSION THREAD

**Discussion by Urfafazli** [Cyfrin/foundry-full-course-cu#2289] [Q&A] [answered]
<img width="1180" alt="Screenshot 2024-08-21 at 10 30 13 PM" src="https://github.com/user-attachments/assets/8f9af6eb-9507-4ee4-a546-ff859b630592">
I am getting while trying to compile as shown in the screenshot. I read different articles and also reached out to AI assistants for help and did everything that was told yet couldn't make it through.

**Comment by Urfafazli** [comment]
I also tried compiling the contract with a specific version installed but that too didn't worked out

**Reply by EngrPips** [reply, marked as answer]
Hello @Urfafazli, Please run `foundryup-zksync` and then rerun the `forge build` command. Can you set your pragma to `0.8.23`? 

<img width="1440" alt="Screenshot 2024-08-21 at 18 13 30" src="https://github.com/user-attachments/assets/d4432d0c-8d63-4674-8575-6ce694a3d0f3">

Solidity version `0.8.23` worked for me.

**Comment by Urfafazli** [comment]
Hey @EngrPips thanks, it worked!!

**Reply by EngrPips** [reply]
Awesome, Keep crushing it buddy.
